session_crypto.py

```
@@ -138,16 +172,21 @@
                 'session_key': session_key,
                 'worker_public_key_data': list(worker_public_key_data)
             }
             
         except Exception as e:
-            logger.error(f"Failed to create session encryption for {session_id[:8]}...: {str(e)}")
-            raise ValueError(f"Session encryption failed: {str(e)}")
+            # Sanitize error message for logging to prevent log injection
+            sanitized_error = sanitize_for_logging(str(e))
+            sanitized_session_id = sanitize_for_logging(str(session_id)[:8])
+            logger.error(f"Failed to create session encryption for {sanitized_session_id}...: {sanitized_error}")
+            raise ValueError("Session encryption failed")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)